src/dispatch.rs

```diff
@@ -290,8 +294,8 @@ impl ApduDispatch
 
                 if self.was_request_chained || res.len() > interchanges::SIZE {
 
-                    // Send 256 bytes max at a time.
-                    let boundary = core::cmp::min(256, res.len());
+                    // Do not send more than the expected bytes
+                    let boundary = core::cmp::min(self.response_len_expected, res.len());
```

Comment:
Hmm, one afterthought:  Shouldn’t we take into account `interchanges::SIZE` here?  `self.response_len_expected` could be up to `u16::MAX + 1`, `res.len()` up to `crate::response::SIZE`.  `interchanges::SIZE` is smaller than that.